packages/service/lib/dependencies.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
+
+const { resolve } = require('node:path')
+
+function getClientId (service, client) {
+  let id
+
+  if (client.path) {
+    try {
+      const clientPackageJsonPath = resolve(service.path, client.path, 'package.json')
+
+      // Let's use require here so we can avoid double parsing
+      id = require(clientPackageJsonPath, 'utf8').name ?? client.id
+    } catch (err) {
+      if (client.url === undefined || client.name === undefined) {
+        throw err
+      }
+    }
+  }
+
+  // This function can also be called for composer services
+  // which don't have a path, so using their id as fallback is reasonable
+  return id ?? client.id
+}
+
+function getServiceUrl (id) {
+  return `http://${id}.plt.local`
+}
+
+async function parseDependency (configManager, id, urlString) {
```

Comment:
This really need to be in `configManager`, `runtime` or `util`, as we will remove the package.json dependency from runtime to service.

Reply:
Moreover, I have a rought feeling this it's not needed here.